{{cookiecutter.repo_name}}/.gitlab-ci.yml

```diff
@@ -0,0 +1,37 @@
+image: alpine
+
+variables:
+  # HTTP_PROXY is used to pull docker images.
+  HTTP_PROXY: {{cookiecutter.ci_https_proxy}}
+  # HTTPS_PROXY is used to install Python packages such as tox.
+  HTTPS_PROXY: {{cookiecutter.ci_https_proxy}}
+  # If we don't include NO_PROXY, we will get "fatal unable to update url base from redirection"
+  # when trying to fetch the gitlab-ci-token.
+  NO_PROXY: 127.0.0.1,localhost,.lan,.local,.home,/var/run/docker.sock,{{cookiecutter.repo_hosting}}
+  # If some of the links in your documentation require a special PEM to verify,
+  # then sphinx -b linkcheck will fail without that PEM.
+  # But setting REQUESTS_CA_BUNDLE to that PEM will cause other links to fail,
+  # because the runner will only accept that PEM, not the defaults.
+  # Therefore you will usually want to bundle all certificates together with
+  # cat `python -c "import requests; print(requests.certs.where())"` ~/your.pem > ~/bundled.pem
+  # REQUESTS_CA_BUNDLE: ci/name-of-your-ca-bundle.pem
+
+pages:
+  script:
+  - apk add python3 git # probably do not want --no-cache
+    # check-manifest, used in tox -e check, requires git.
+  - pip3 install --upgrade pip
+  - pip3 install tox
+  - git --version
+  - python3 --version
+  - virtualenv --version
+  - pip3 --version
+  - uname --all
+  # Some distributions do not include the standard lsb_release command.
+  - tox
+  - mv dist/docs/ public/
+  artifacts:
+    paths:
+    - public
```

Comment:
I guess you might want to put other stuff in public? You could simply have `paths: - dist`?

Reply:
The documentation doesn't really make this clear, but for GitLab Pages the folder name *must* be "public"; otherwise you get weird errors. See e.g. https://gitlab.com/gitlab-org/gitlab-ce/issues/40686

Hmm. It might be worth putting a comment in .gitlab-ci.yml warning users "yes it sure does *look* like you could change this to anything you want, but you can't." I expect most people looking at .gitlab-ci.yml would immediately assume, just like you did, that they could change the path to be anything.